src/libraries/Common/src/System/Runtime/InteropServices/BuiltInVariantExtensions.cs

```diff
@@ -146,7 +144,7 @@ public static unsafe void CopyFromIndirect(this ref ComVariant variant, object v
                 VarEnum.VT_UI8 => variant.As<ulong>(),
                 VarEnum.VT_INT => variant.As<int>(),
                 VarEnum.VT_UINT => variant.As<uint>(),
-                VarEnum.VT_BOOL => variant.As<short>() != -1,
+                VarEnum.VT_BOOL => variant.As<short>() != ComVariant.VARIANT_FALSE,
```

Comment:
Always compare to `VARIANT_TRUE`.

Reply:
Is it the recommended pattern for `VARIANT_BOOL`?

The recommended pattern for `BOOL` is to compare against FALSE and native code static analysis tools will complain when it is compared against `TRUE`. I would expect that the recommended pattern for `VARIANT_BOOL` is going to be the same.

Reply:
I'm seeing more guidance to compare against VARIANT_FALSE instead of VARIANT_TRUE as that better catches "incorrect" cases from misbehaving APIs (as `false` and `VARIANT_FALSE` are the same value, whereas `VARIANT_TRUE` and `true` are different bit patterns)

Reply:
Historically on the VS team, the comparrison has always been against `VARIANT_TRUE` because that is the only "true" value. Non-zero values, other than `-1` are not true, but errors. This was the impetus for the statement. However, @jkoritzinsky pointed out that the runtime current compares against `VARIANT_FALSE`, which I find odd but if those are the semantics we want to accept then so be it.